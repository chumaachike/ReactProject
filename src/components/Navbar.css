.nav-bar {
  width: 80%;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 40px;
  padding: 0 10px;
}

header{
  padding: 28px 0px;
  width: 100%;
  background-color: #9da5bd;
}

.logo-image {
  width: 5%;
  height: 5%;
}

.logo-brand {
  display: flex;
  gap: 10px;
  align-items: center;
}

.nav-brand a{
  font-family: "Montserrat", sans-serif;
  font-size: 1.875rem;
  font-weight: bold;
  color: #000;
  text-decoration: none;
}

.nav-links {
  display: flex;
  list-style-type: none;
  gap: 20px;
}

.nav-link a{
  color: 	#040c24;
  font-weight: 600;
  font-size: 21px;
  text-decoration: none;
}

li .active {
  font-weight: bold;
  text-decoration: underline;
}