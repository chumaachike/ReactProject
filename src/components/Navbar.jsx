import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../planet.svg';
import './Navbar.css';

function Navbar() {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/mission',
      text: 'Mission',
    },
    {
      id: 3,
      path: '/profile',
      text: 'Profile',
    },
  ];
  return (
    <header>
      <nav className="nav-bar">
        <div className="logo-brand">
          <img src={logo} alt="planet logo" className="logo-image" />
          <h2 className="nav-brand"><a href="/">Space Travelers&apos; Hub</a></h2>
        </div>
        <ul className="nav-links">
          {links.map((link) => (
            <li className="nav-link" key={link.id}>
              <NavLink className={(navData) => (navData.isActive ? 'active' : 'none')} to={link.path}>
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Navbar;
