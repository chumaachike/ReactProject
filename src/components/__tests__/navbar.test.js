import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

describe('snapshot test for Navbar component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot('navbar');
  });
});
