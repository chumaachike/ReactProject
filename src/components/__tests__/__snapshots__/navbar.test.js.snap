// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Navbar component renders correctly: navbar 1`] = `
<header>
  <nav
    className="nav-bar"
  >
    <div
      className="logo-brand"
    >
      <img
        alt="planet logo"
        className="logo-image"
        src="planet.svg"
      />
      <h2
        className="nav-brand"
      >
        <a
          href="/"
        >
          Space Travelers' Hub
        </a>
      </h2>
    </div>
    <ul
      className="nav-links"
    >
      <li
        className="nav-link"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          className="none"
          href="/mission"
          onClick={[Function]}
        >
          Mission
        </a>
      </li>
      <li
        className="nav-link"
      >
        <a
          className="none"
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
